### PR TITLE
Add customizable ban tokens

### PR DESCRIPTION
**Pull request by sALTaccount**
## Checklist:

- [ X ] I have read the [Contributing guidelines](https://github.com/oobabooga/text-generation-webui/wiki/Contributing-guidelines).

This PR adds a feature where you can input token IDs that you want to ban from generating. It uses the same logic as the EOS token ban. This feature is useful for controlling the generations of models. For example, I wanted to prevent the model from describing the actions in RP such as `*looks down*`, so I put all tokens IDs containing a `*` in the ban box, and the model can no longer generate those actions.

I've tested that this works for all exllama (including HF ones), autoGPTQ, GPTQ for llama, and llama.cpp. I've also tested that it is accessible via the API

### CODE REVIEW

**Comment by Ph0rk0z** [comment]
oh boy! finally logits banning

**Comment by BadisG** [comment]
How do we find the ID of the token we want to ban though?

**Comment by BadisG** [comment]
I tried it and it works fine, good job.
Maybe it would be a bit more convenient to have the option to save the tokens we want to ban, like having it on the presets save or something like that? I'm not a big fan of keeping all those ID in a notebook and then paste it on the webui everytime I want to run it.

**Comment by sALTaccount** [comment]
Token bans can now be defined by presets. Didn't realize that wasn't working earlier

**Comment by sALTaccount** [comment]
@oobabooga what do you think of this PR?

**Comment by oobabooga** [comment]
I think it's perfect. It's nice that you dug through each loader, including the non _HF ones, and the PR is very clean overall. Thank you!
